cas-server-core/src/main/java/org/jasig/cas/util/CompressionUtils.java

```diff
@@ -60,13 +60,29 @@ private CompressionUtils() {
      * @return the array as a string with <code>UTF-8</code> encoding
      */
     public static String inflate(final byte[] bytes) {
-        try (ByteArrayInputStream inb = new ByteArrayInputStream(bytes);
-             ByteArrayOutputStream out = new ByteArrayOutputStream();
-             InflaterOutputStream ios = new InflaterOutputStream(out);) {
-            IOUtils.copy(inb, ios);
-            return new String(out.toByteArray(), UTF8_ENCODING);
-        } catch (final Exception e) {
-            throw new RuntimeException(e.getMessage(), e);
+        final Inflater inflater = new Inflater(true);
```

Comment:
Here is the fix, isn't it? It looks like we returned to the first implementation we had for that and which was replaced by the `InflaterOutputStream `.